python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -269,6 +296,13 @@ def __init__(
 
         self._num_readers = len(self._readers)
         if self.is_remote():
+            from ray.dag.context import GRPC_MAX_PAYLOAD
+
+            if typ.buffer_size_bytes > GRPC_MAX_PAYLOAD:
+                raise ValueError(
```

Comment:
Maybe we should say that the `GRPC_MAX_PAYLOAD` constraint only applies to the case where the reader and writer are on different nodes?

Reply:
Done. I updated the exception message.